docs/source/posts/2024/2024-06-06-week-1-robin.rst

```diff
@@ -0,0 +1,106 @@
+
+
+Things I did week-1 (and some week2)
+------------------------------------
+
+1) **Choosing the vector database**
+
+I decided to choose `Pinecone <https://www.pinecone.io/>`_ as the vector DB because it had a very generous free tier. Other options on consideration were `pgvector <https://github.com/pgvector/pgvector>`_ and `chromadb <https://www.trychroma.com/>`_, but they didn't have a free tier.
+
+2) **PR Submissions and Review**
+
+I also merged a `PR <https://github.com/fury-gl/fury/pull/891>`_ on FURY which fixes a CI issue. I also spent time doing review of other PRs from my fellow GSoC mates.
+
+3) **Deciding which embedding model to use**
+
+A good embedding model is necessary to generate embeddings which we then upsert into the DB. Ollama had embedding model support, but I found the catalogue very small and the models they provided were not powerful enough. Therefore I decided to try using HuggingFace Sentence Transformers.
+Sentence Transformers have a very vibrant catalogue of models available of various sizes. I chose `gte-large-en-v1.5 <https://huggingface.co/Alibaba-NLP/gte-large-en-v1.5>`_ from Alibaba-NLP, an 8k context, 434 million parameter model. It only had a modest memory requirement of 1.62 GB.
+Performance wise, it ranks 11th on the `MTEB leaderboard <https://huggingface.co/spaces/mteb/leaderboard>`_. It is a very interesting model due to its size:performance ratio.
+
+4) **Hosting the embedding model**
+
+Hosting this sentence-transformer model was confusing. For some reason, the HF spaces were blocking the Python script from writing on ``.cache`` folder. Later I figured out the reason and fixed it.
+I've hosted 5 gunicorn workers to serve 5 parallel requests at a time. Since the model is small, this is possible.
+
+5) **Hosting the database endpoint**
+
+I wrapped the pinecone db api into an endpoint so it'll be easy to query and receive the results.
+It is also configured to accept 5 concurrent requests although I could increase it a lot more.
+
+I upserted docstrings from ``fury/actor.py`` into the vector db for testing. So now, whenever you ask a question the model will use some ``actor.py`` function to give you an answer. For now, it could be used like a semantic function search engine.
+
+I decided to abstract the db endpoint to reduce the dependency on one provider. We can swap the providers as required and keep all other features running.
+
+6) **Hosting Discord Bot**
+
+So with this, all the endpoints are finally online. The bot has some issues, it is going offline midway for some reason. I'll have to see why that happens.
+
+For some reason, huggingface spaces decided to not start the bot script. Later a community admin from huggingface told me to use their official bot implementation as a reference. This is why I had to use threading and gradio to get the bot running (migrating to docker can be done, but this is how they did it and I just took that for now).
```

Comment:
Also, try to be consistent with how you write other site/organization names, for correctness. Here I refer to `huggingface -> Hugging Face` or `HuggingFace`